### DISCUSSION TITLE
Is the mounted directory the reason for Tart's near-native performance?

### DISCUSSION THREAD

**Discussion by timcotterell** [cirruslabs/tart#420] [Q&A] [answered]
I have been trying to determine the best option for virtualization on my M1 MacBook Pro, and I have tested every virtualization offering I could find. I found a huge discrepancy between performance of Tart versus every other flavor. All the other virtualization platforms I looked into suffer about a 30-40% performance degradation when building and testing my xcode projects, but Tart suffers less than 5%. I went through all the platforms and the only difference I have seen is that the other platforms had the storage integrated with the image, and tart prefers a mounted directory. I just wanted to confirm that this is the reason for the vast difference I see in performance between Tart and other M1 virtualization offerings.

**Comment by fkorotkov** [comment, marked as answer]
Yes, IO for mounted directories is on par with the host where the integrated disk storage is about 30% slower due to the extra abstraction. If you'd like more in-depth details, here is a [very insightful talk](https://stream.lib.utah.edu/index.php?c=details&id=13535) where at 11:15 Howard is touching the disk performance.
